src/collection_manager.cpp

```diff
@@ -531,8 +531,17 @@ Option<nlohmann::json> CollectionManager::drop_collection(const std::string& col
     std::unique_lock u_lock(mutex);
     collections.erase(actual_coll_name);
     collection_id_names.erase(collection->get_collection_id());
+
+
     u_lock.unlock();
 
+    const auto& embedding_fields = collection->get_embedding_fields();
```

Comment:
We still need a lock on this because we don't want another operation to modify embedding fields before we are done looping through it.